bigframes/core/rewrite.py

```diff
@@ -113,10 +111,17 @@ def maybe_join(
         if (self.root != right.root) or any(
             l_expr != r_expr for l_expr, r_expr in zip(l_join_exprs, r_join_exprs)
         ):
+            return False
+        return True
+
+    def merge(
```

Comment:
maybe_merge? I'd like to telegraph that this can return None.

Reply:
done